### PR TITLE
Add raymath and raygui support through the new Parser

### LINKED ISSUE #145: Update Parser
[raylib_api.json](https://github.com/raysan5/raylib/blob/master/parser/raylib_api.json) now includes...
- Aliases
- `FLOAT_MATH` instead of `UNKNOWN`
- Actual Matrix

### PR DESCRIPTION
**Pull request by RobLoach**
Given that the parser has changed, and it's difficult for us to target different versions of the API vs parser version, it may be worth it to manage it ourselves. The provided `raylib_api.json` was built using the latest Parser Generator from `master`, with the `4.0.0` raylib.h.

DO NOT MERGE YET! It's not compiling with this yet. We'll need to wrap...

- [ ] `float[4]`
- [ ] `char[32]`
- [ ] `Matrix[2]`
- [ ] `float[2]`
- [ ] `float[16]`
- [ ] Aliases

Fixes #145

### CODE REVIEW

File: src/generated/node-raylib.js [hunk 1]
```diff
@@ -457,20 +457,20 @@ raylib.SetShaderValueMatrix = function (shader, locIndex, mat) {
...
-    mat.m5,
-    mat.m6,
-    mat.m7,
     mat.m8,
-    mat.m9,
-    mat.m10,
-    mat.m11,
     mat.m12,
+    mat.m1,
+    mat.m5,
```
Comment (RobLoach) [inline src/generated/node-raylib.js:464]: Something up with the Matrix field sorting

File: tools/generate.js [hunk 2]
```diff
@@ -25,8 +25,19 @@ const blocklist = [
...
+  // @todo Verify Structs that have array properties
+  'VrStereoConfig', // Matrix[2], float[2]
+  'BoneInfo', // char[32]
+  'VrDeviceInfo', // float[4]
+  'Material', // float[4]
+
+  // raymath
+  // @todo Fix helper structs for float arrays
+  'Vector3ToFloatV', // float3
+  'MatrixToFloatV' // float16
```
[the remark this answers is not in the document]
Comment (twuky) [reply tools/generate.js:40]: so I tried to do some work on this front - but ended up throwing out the changes. my understanding of types there was a little wrong. looking at the structs in question - i don't see a need to implicitly cast to/from a JS array anywhere for these unless a user really wanted to manipulate this data manually. I think we can get by with simply casting the pointers for these types into uint64_t, and then (float[]*)* or what the proper syntax might be on the way into C++. 
for Matrix/Vector3V - can these just be implemented in JS? or does raylib do anything extra under the hood when making these?

[the remark this answers is not in the document]
Comment (RobLoach) [reply tools/generate.js:40]: Maybe be difficult to handle directly through the generator. We could likely put together wrapper functions for them though. Similar to the Shader ones you did.

File: package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "start": "bin/node-raylib examples/core/core_basic_window.js",
     "benchmark": "bin/node-raylib examples/textures/textures_bunnymark.js",
     "test": "jest",
-    "posttest": "standard tools test examples index.js",
+    "posttest": "#standard tools test examples index.js",
```
[the remark this answers is not in the document]
Comment (RobLoach) [reply package.json:11]:
``` suggestion
    "posttest-ignore": "standard tools test examples index.js",
```

File: package.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
     "start": "bin/node-raylib examples/core/core_basic_window.js",
     "benchmark": "bin/node-raylib examples/textures/textures_bunnymark.js",
     "test": "jest",
-    "posttest": "standard tools test examples index.js",
+    "posttest-ignore": "standard tools test examples index.js",
```
Comment (RobLoach) [inline package.json:11]: Need to restore the `standard` test once the tests pass.

File: tools/generate_templates/node-raylib-bindings.js [hunk 5]
```diff
@@ -211,8 +234,15 @@ inline unsigned char unsignedcharFromValue(const Napi::CallbackInfo& info, int i
...
-  return info[index].As<Napi::Number>();
+  return info[index].As<Napi::Number>().Uint32Value();
 }
+// inline float* float4FromValue(const Napi::CallbackInfo& info, int index) {
+//   Napi::Float32Array array = info[index].As<Napi::Float32Array>();
+//   size_t size = array.ElementLength();
+//   float* buffer = new float[size];
+//   memcpy(buffer, array.Data(), size);
+//   return buffer;
+// }
```
Comment (RobLoach) [inline tools/generate_templates/node-raylib-bindings.js:245]:
``` suggestion
```

File: tools/generate.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (konsumer) [reply tools/generate.js:40]: Could we do it in js wrapper-space? Like how ref-array-napi does it? https://www.npmjs.com/package/ref-array-napi

File: package.json (hunk 4 above)
Comment (konsumer) [reply package.json:11]: `standard --fix` can be helpful, but I also use it in my editor for automatic fix-on-0save

**Comment by RobLoach** [comment]
[ 96%] Building CXX object CMakeFiles/node-raylib.dir/src/generated/node-raylib.cc.o
/Users/runner/work/node-raylib/node-raylib/src/generated/node-raylib.cc:268:6: error: non-constant-expression cannot be narrowed from type 'uintptr_t' (aka 'unsigned long') to 'char' in initializer list [-Wc++11-narrowing]
     pointerFromValue(info, index + 0),
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/runner/work/node-raylib/node-raylib/src/generated/node-raylib.cc:268:6: note: insert an explicit cast to silence this issue
     pointerFromValue(info, index + 0),
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     static_cast<char>(               )
1 error generated.
make[2]: *** [CMakeFiles/node-raylib.dir/src/generated/node-raylib.cc.o] Error 1
make[1]: *** [CMakeFiles/node-raylib.dir/all] Error 2
make: *** [all] Error 2
info REP Build has been failed, trying to do a full rebuild.

**Comment by konsumer** [comment]
I'm not sure I totally follow what changes are being made and what is failing, but I'd like to help. Is the issue that the generated code uses `uintptr_t` which is 64bit, but char is 1 byte? Is this what is intended? If this is what we intend, can we just do a cast there?

**Comment by konsumer** [comment]
I got it to build by hand-editing this:

```cpp
inline BoneInfo BoneInfoFromValue(const Napi::CallbackInfo& info, int index) {
  return {
     (char) pointerFromValue(info, index + 0),
     charFromValue(info, index + 1)
  };
}
```

instead of ```cpp
inline BoneInfo BoneInfoFromValue(const Napi::CallbackInfo& info, int index) {
  return {
     pointerFromValue(info, index + 0),
     charFromValue(info, index + 1)
  };
}
```

Not sure if that will actually work right, but might lead to a solution.

**Comment by konsumer** [comment]
I added a small change that gets it building in a542777 but I'm really not sure if that is the right way to go. Seems like it probly isn't dealing with arrays right, but I'm not sure how to test it.

**Pull request merged**